From f57ec90b0d3a93dc7ed1b3e08ff5c6d7179280e7 Mon Sep 17 00:00:00 2001
From: Justin Bronder <jsbronder@gmail.com>
Date: Thu, 3 Mar 2016 10:43:47 -0500
Subject: [PATCH] librc: sh:  respect alternative INITDIR

Remove hard-coded directory name for INITDIR, which was partially
respected by the build system but was being ignored at runtime.
---
 mk/scripts.mk       | 2 +-
 sh/gendepends.sh.in | 2 +-
 src/librc/Makefile  | 2 ++
 src/librc/rc.h.in   | 3 ++-
 4 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/mk/scripts.mk b/mk/scripts.mk
index 0cbd3bf..e08b213 100644
--- a/mk/scripts.mk
+++ b/mk/scripts.mk
@@ -19,7 +19,7 @@ _PKG_SED:=		$(shell ${_PKG_SED_SH})
 _LCL_SED_SH=		if test "${PREFIX}" = "${LOCAL_PREFIX}"; then echo "-e 's:@LOCAL_PREFIX@::g'"; else echo "-e 's:@LOCAL_PREFIX@:${LOCAL_PREFIX}:g'"; fi
 _LCL_SED:=		$(shell ${_LCL_SED_SH})
 
-SED_REPLACE=		-e 's:@SHELL@:${SH}:g' -e 's:@LIB@:${LIBNAME}:g' -e 's:@SYSCONFDIR@:${SYSCONFDIR}:g' -e 's:@LIBEXECDIR@:${LIBEXECDIR}:g' -e 's:@PREFIX@:${PREFIX}:g' -e 's:@BINDIR@:${BINDIR}:g' -e 's:@SBINDIR@:${SBINDIR}:g' ${_PKG_SED} ${_LCL_SED}
+SED_REPLACE=		-e 's:@SHELL@:${SH}:g' -e 's:@LIB@:${LIBNAME}:g' -e 's:@SYSCONFDIR@:${SYSCONFDIR}:g' -e 's:@LIBEXECDIR@:${LIBEXECDIR}:g' -e 's:@PREFIX@:${PREFIX}:g' -e 's:@BINDIR@:${BINDIR}:g' -e 's:@SBINDIR@:${SBINDIR}:g' -e 's:@INITDIR@:${INITDIR}:g' ${_PKG_SED} ${_LCL_SED}
 
 # Tweak our shell scripts
 %.sh: %.sh.in
diff --git a/sh/gendepends.sh.in b/sh/gendepends.sh.in
index 82088c5..e05b1e4 100644
--- a/sh/gendepends.sh.in
+++ b/sh/gendepends.sh.in
@@ -55,7 +55,7 @@ depend() {
 
 _done_dirs=
 for _dir in \
-@SYSCONFDIR@/init.d \
+@INITDIR@ \
 @PKG_PREFIX@/etc/init.d \
 @LOCAL_PREFIX@/etc/init.d
 do
diff --git a/src/librc/Makefile b/src/librc/Makefile
index 08c599e..cdf6a93 100644
--- a/src/librc/Makefile
+++ b/src/librc/Makefile
@@ -20,6 +20,8 @@ SED_CMD+=	-e 's:@SYSCONFDIR@:${SYSCONFDIR}:g'
 SED_CMD+=	-e 's:@LIBEXECDIR@:${LIBEXECDIR}:g'
 SED_CMD+=	-e 's:@BINDIR@:${BINDIR}:g'
 SED_CMD+=	-e 's:@SBINDIR@:${SBINDIR}:g'
+SED_CMD+=	-e 's:@INITDIR@:${INITDIR}:g'
+SED_CMD+=	-e 's:@CONFDIR@:${CONFDIR}:g'
 
 _PKG_PREFIX=	-e 's:.*@PKG_PREFIX@.*:\#undef RC_PKG_PREFIX:g'
 ifneq (${PKG_PREFIX},)
diff --git a/src/librc/rc.h.in b/src/librc/rc.h.in
index 92ecbb4..4f6f64e 100644
--- a/src/librc/rc.h.in
+++ b/src/librc/rc.h.in
@@ -26,6 +26,8 @@ extern "C" {
 #define RC_SYSCONFDIR		"@SYSCONFDIR@"
 #define RC_LIBDIR               "@PREFIX@/@LIB@/rc"
 #define RC_LIBEXECDIR           "@LIBEXECDIR@"
+#define RC_INITDIR              "@INITDIR@"
+
 #if defined(PREFIX)
 #define RC_SVCDIR               RC_LIBEXECDIR "/init.d"
 #elif defined(__linux__) || (defined(__FreeBSD_kernel__) && \
@@ -35,7 +37,6 @@ extern "C" {
 #define RC_SVCDIR               RC_LIBEXECDIR "/init.d"
 #endif
 #define RC_RUNLEVELDIR          RC_SYSCONFDIR "/runlevels"
-#define RC_INITDIR              RC_SYSCONFDIR "/init.d"
 #define RC_CONFDIR              RC_SYSCONFDIR "/conf.d"
 #define RC_PLUGINDIR            RC_LIBDIR "/plugins"
 
-- 
2.4.10

